tools/floodgate/js/floodgate.js

```diff
@@ -12,15 +22,116 @@ import {
+let project;
+let config;
+
 async function reloadProject() {
   loadingON('Purging project file cache and reloading... please wait');
   await purgeAndReloadProjectFile();
 }
 
+async function refreshPage() {
+  // Inject Sharepoint file metadata
+  loadingON('Updating Project with the Sharepoint Docs Data...');
+  await updateProjectWithDocs(projectDetail);
+
+  // Render the data on the page
+  loadingON('Updating tabeke with project details..');
+  await updateProjectDetailsUI(projectDetail, config);
+  loadingON('UI updated..');
+  loadingOFF();
+}
+
+async function floodgateContent() {
+  function updateAndDisplayCopyStatus(copyStatus, srcPath) {
+    let copyDisplayText = `Copied ${srcPath} to floodgated content folder`;
+    if (!copyStatus) {
+      copyDisplayText = `Failed to copy ${srcPath} to floodgated content folder`;
+    }
+    loadingON(copyDisplayText);
+  }
+
+  async function copyFilesToFloodgateTree(urlInfo) {
+    const status = { success: false };
+    try {
+      const srcPath = urlInfo?.doc?.filePath;
+      loadingON(`Copying ${srcPath} to pink folder`);
+      let copySuccess = false;
+      if (urlInfo?.doc?.fg?.sp?.status !== 200) {
+        const destinationFolder = `${srcPath.substring(0, srcPath.lastIndexOf('/'))}`;
+        copySuccess = await copyFile(srcPath, destinationFolder, undefined, true);
+        updateAndDisplayCopyStatus(copySuccess, srcPath);
+      } else {
+        // Get the source file
+        const file = await getFile(urlInfo.doc);
+        if (file) {
+          const destination = urlInfo?.doc?.filePath;
+          if (destination) {
+            // Save the file in the floodgate destination location
+            const saveStatus = await saveFile(file, destination, true);
+            if (saveStatus.success) {
+              copySuccess = true;
+            }
+          }
+        }
+        updateAndDisplayCopyStatus(copySuccess, srcPath);
+      }
+      status.success = copySuccess;
+      status.srcPath = srcPath;
+      status.dstPath = srcPath;
+      status.url = urlInfo.doc.url;
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.log(`Error occurred when trying to copy files to floodgated content folder ${error.message}`);
+    }
+    return status;
+  }
+
+  hideButtons(ACTION_BUTTON_IDS);
+  const startCopy = new Date();
+  const copyStatuses = await Promise.all(
+    [...projectDetail.urls].map(((valueArray) => copyFilesToFloodgateTree(valueArray[1]))),
+  );
+  const endCopy = new Date();
+
+  loadingON('Previewing for copied files... ');
+  const previewStatuses = await Promise.all(
+    copyStatuses
+      .filter((status) => status.success)
+      .map((status) => simulatePreview(stripExtension(status.dstPath), 1, true)),
+  );
+  loadingON('Completed Preview for copied files... ');
+
+  const failedCopies = copyStatuses
+    .filter((status) => !status.success)
+    .map((status) => status?.srcPath || 'Path Info Not available');
```

Comment:
Not sure about the placement of the optional chaining operator here. If status was nullish, wouldn't it error out when you try to get `status.success`